src/Controller/MailPreviewController.php

```diff
@@ -0,0 +1,275 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.3
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace DebugKit\Controller;
+
+use Cake\Collection\CollectionInterface;
+use Cake\Controller\Controller;
+use Cake\Core\App;
+use Cake\Core\Configure;
+use Cake\Core\Plugin;
+use Cake\Event\Event;
+use Cake\Network\Exception\NotFoundException;
+use Cake\Utility\Inflector;
+use DebugKit\Mailer\AbstractResult;
+use DebugKit\Mailer\PreviewResult;
+use DebugKit\Mailer\SentMailResult;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Provides access to the MailPreview classes for visually debugging email sending
+ *
+ */
+class MailPreviewController extends Controller
+{
+    /**
+     * Before filter callback.
+     *
+     * @param \Cake\Event\Event $event The beforeFilter event.
+     * @return void
+     * @throws Cake\Network\Exception\NotFoundException
+     */
+    public function beforeFilter(Event $event)
+    {
+        if (!Configure::read('debug')) {
+            throw new NotFoundException();
+        }
+    }
+
+    /**
+     * Before render handler.
+     *
+     * @param \Cake\Event\Event $event The event.
+     * @return void
+     */
+    public function beforeRender(Event $event)
+    {
+        $this->viewBuilder()->layout('DebugKit.mailer');
+    }
+
+    /**
+     * Handles mail-preview/index
+     *
+     * @return void
+     */
+    public function index()
+    {
+        $this->set('mailPreviews', $this->getMailPreviews()->toArray());
+    }
+
+    /**
+     * Handles the viewing of an already sent email that was logged in the Mail panel
+     * for DebugKit
+     *
+     * @param string $panelId The Mail panel id where the email data is stored
+     * @param string $number The email number as stored in the logs
+     * @return \Psr\Http\Message\ResponseInterface|null
+     */
+    public function sent($panelId, $number)
+    {
+        $this->loadModel('DebugKit.Panels');
+        $panel = $this->Panels->get($panelId);
+        // @codingStandardsIgnoreStart
+        $content = @unserialize($panel->content);
+        // @codingStandardsIgnoreEnd
+
+        if (empty($content['emails'][$number])) {
+            throw new NotFoundException('No emails found in this request');
+        }
+
+        $email = $content['emails'][$number];
+        $email = new SentMailResult(array_filter($email['headers']), $email['message']);
+        $partType = $this->request->query('part');
```

Comment:
Nitpicking: Move this one line below (moving the whiteline one up, so the assignment and the check are together.